### PR TITLE
14.0 purchase quick improvements

### PR DESCRIPTION
**Pull request by kevinkhao**
- include https://github.com/OCA/purchase-workflow/pull/1009/files
- add functionality on UoMs: in addition to quantity, select the appropriate UoM
- refactored tests
~This is WIP, some things must be moved around to the base module base_product_mass_addition~

### CODE REVIEW

**Comment by kevinkhao** [comment]
~depends on https://github.com/akretion/product-attribute/pull/12~
see below for updated PR dependency

File: purchase_quick/models/product_product.py [hunk 1]
```diff
@@ -19,21 +19,34 @@ class ProductProduct(models.Model):
...
+            for rec in self:
+                rec.quick_uom_category_id = rec.uom_po_id.category_id
+        else:
+            super()._compute_quick_uom_category_id()
+
+    def _compute_process_qty_purchase(self):
+        po_lines = self.env["purchase.order.line"].search(
+            [("order_id", "=", self.env.context.get("parent_id"))]
+        )
+        for product in self:
```
Comment (sebastienbeau) [inline purchase_quick/models/product_product.py:31]: @PierrickBrun @kevinkhao 
I think we have an issue here (the issue was here before).
We sum the qty of product for each line. But them we write the total qty in the first line.

So in the case I have 2 PO line with product FOO:
- 10 x FOO
- 20 x FOO

I will have 30 FOO in the product qty, If I set 35 I will have on my PO
- 35 FOO
- 20 FOO

In order to keep this module simplier, and also stick to real case.
I think we should keep the sum here BUT add a raise in _get_quick_line if we have several line.

In real word we should never have twice the same product in a PO and If it's the case, for example with purchase_lot, it doesn't make sense to use mass addition.

So let's keep it simple and not broken

Comment (kevinkhao) [reply purchase_quick/models/product_product.py:31]: done

Comment (kevinkhao) [reply purchase_quick/models/product_product.py:31]: done

**Comment by kevinkhao** [comment]
@sebastienbeau modified & automatic & manual tested

It seems travis cannot find the right branch (package not found?) I think https://github.com/akretion/product-attribute/pull/12 must be merged first

File: purchase_quick/tests/test_quick_purchase.py [hunk 2]
```diff
@@ -1,67 +1,105 @@
...
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo.tests.common import Form, TransactionCase
+from odoo.exceptions import ValidationError
+from odoo.tests.common import Form, SavepointCase
 
 
-class TestQuickPurchase(TransactionCase):
+class TestQuickPurchase(SavepointCase):
+    # TODO is the below still relevant ?
```
Comment (PierrickBrun) [inline purchase_quick/tests/test_quick_purchase.py]: Can you remove the below text (I think this is a copy/pasting error)

**Review by PierrickBrun — APPROVED** [review]

**Comment by kevinkhao** [comment]
Now depends on https://github.com/OCA/product-attribute/pull/864
(same branch, different PR)

File: test-requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+odoo14-addon-base_product_mass_addition @ git+https://github.com/OCA/product-attribute@refs/pull/864/head#subdirectory=setup/base_product_mass_addition
```
Comment (sebastienbeau) [inline test-requirements.txt]: remove this when PR is merged : https://github.com/OCA/product-attribute/pull/864

File: oca_dependencies.txt [hunk 4]
```diff
@@ -1,2 +1,2 @@
 server-ux
-product-attribute https://github.com/akretion/product-attribute 14.0-mig-product_base_mass_addition
```
Comment (sebastienbeau) [inline oca_dependencies.txt:2]: remove specific branch here when https://github.com/OCA/product-attribute/pull/864 is merged

**Review by bguillot — APPROVED** [review]
LGTM Tested

**Review by sebastienbeau — APPROVED** [review]
LGTM

**Pull request merged**